src/SignalR/server/Core/src/HubOptionsSetup.cs

```diff
@@ -26,6 +27,10 @@ public HubOptionsSetup(IEnumerable<IHubProtocol> protocols)
         {
             foreach (var hubProtocol in protocols)
             {
+                if (hubProtocol.GetType().CustomAttributes.Where(a => a.AttributeType.Name == "NonDefaultHubProtocol").Any())
```

Comment:
Check the full type name, namespace included. The thing that's allowed to vary is the *assembly*.

Reply:
Not a huge fan. But if we did this, we should move it to... Microsoft.AspNetCore.SignalR.Internal.NonDefaultHubProtocol?

Reply:
What's the point of verifying the full type name?

Are we worried that someone else might accidentally create their own `NonDefaultHubProtocolAttribute` in a different namespace for a different purpose? And if we worried about preventing people doing it on purpose, shouldn't we verify the type comes from the expected assembly as well?

Reply:
> What's the point of verifying the full type name?

Well, it's the same way this pattern is done elsewhere, particularly [in the compiler](https://github.com/dotnet/roslyn/blob/077df2adbbf3d8dc8dbc2ec60024e5fa78234b92/src/Compilers/Core/Portable/Symbols/Attributes/AttributeDescription.cs#L427) which is the pattern I was intending to emulate with the suggestion. It feels like we should be justifying a change from that pattern if we want to do so. What's the concern with adding the full namespace qualification here? If this attribute were public, it would be `Microsoft.AspNetCore.SignalR.NonDefaultHubProtocolAttribute`, no?

> Not a huge fan. But if we did this, we should move it to... Microsoft.AspNetCore.SignalR.Internal.NonDefaultHubProtocol?

Fine with me. It is, in theory, a "pubternal" attribute.